Week2/homework.js

```diff
@@ -0,0 +1,164 @@
+  const result = await runQuery(
+    `
+    SELECT DISTINCT
+      countrylanguage.language
+    FROM
+      country,
+      countrylanguage
+    WHERE
+      countrylanguage.countrycode = country.code
+        AND country.region = ?
+    ORDER BY language`,
+    regionName,
+  );
+  if (!result.length) throw new Error('Wrong region name');
+  console.log(`The languages spoken in ${regionName} : ${result.map(c => c.language).toString()}`);
+};
+
+//*Find the number of cities in which language Z is spoken (Accept Z from user)
+const getNumberOfSpokenCities = async () => {
+  const question = [
+    {
+      type: 'input',
+      name: 'language',
+      message: 'Language?',
+    },
+  ];
+  const { language } = await inquirer.prompt(question);
+  const result = await runQuery(
+    `
+    SELECT 
+      countrylanguage.language, COUNT(city.name) numberOfCities
+    FROM
+      country,
+      city,
+      countrylanguage
+    WHERE
+      country.code = city.countrycode
+        AND country.code = countrylanguage.countrycode
+        AND countrylanguage.language = ?
+    GROUP BY language;
+    `,
+    language,
+  );
+  if (!result.length) throw new Error('Wrong language');
+  console.log(`The count of ${language} spoken cities  : ${result[0].numberOfCities}`);
+};
+
+//* Accept the region and language from the user. Are there any countries in this region with the given language as the official language ? If yes, display those countries. If no, display FALSE. E.g. (A) input region : 'Western Europe' and input language : 'Dutch' output should be Belgium and Netherlands (B) input region : 'Western Europe' and input language : 'Hindi' output should be 'FALSE'
+const getCountriesOfSpokenLangs = async () => {
+  const question = [
+    {
+      type: 'input',
+      name: 'regionName',
+      message: 'Region name?',
+    },
+    {
+      type: 'input',
+      name: 'language',
+      message: 'Language?',
+    },
+  ];
+  const { regionName, language } = await inquirer.prompt(question);
+  const result = await runQuery(
+    `
+    SELECT DISTINCT
+      country.name country
+    FROM
+      country,
+      countrylanguage
+    WHERE
+      countrylanguage.countrycode = country.code
+        AND country.region = ?
+        AND countrylanguage.language = ?
+    ORDER BY country`,
+    [regionName, language],
+  );
+  console.log(
+    result.length
+      ? `The ${language} spoken countries in ${regionName} : ${result
+          .map(c => c.country)
+          .toString()}`
+      : 'FALSE',
+  );
+};
+
+//* List all the continents with the number of languages spoken in each continent
+const getNumberOfLangsByContinent = async () => {
+  const result = await runQuery(
+    `
+    SELECT 
+      country.continent,
+    COUNT(countrylanguage.language) AS numberOfLanguages
+    FROM
+      country,
+      countrylanguage
+    WHERE
+      country.code = countrylanguage.countrycode
+    GROUP BY continent
+    ORDER BY numberOfLanguages DESC`,
+  );
+
```

Comment:
What is english is spoken by more than one country in america ?

Reply:
I missed repeating languages for a continent the query must be like with `distinct countrylanguage.language`

```
SELECT 
      country.continent,
    COUNT(distinct countrylanguage.language) AS numberOfLanguages
    FROM
      country,
      countrylanguage
    WHERE
      country.code = countrylanguage.countrycode
    GROUP BY continent
    ORDER BY numberOfLanguages DESC
```